src/sinks/hdfs/config.rs

```diff
@@ -0,0 +1,190 @@
+    sinks::{
+        opendal_common::*,
+        util::{
+            partitioner::KeyPartitioner, BatchConfig, BulkSizeBasedDefaultBatchSettings,
+            Compression,
+        },
+        Healthcheck,
+    },
+};
+
+/// Configuration for the `hdfs` sink.
+///
+/// The Hadoop Distributed File System (HDFS) is a distributed file system
+/// designed to run on commodity hardware. HDFS consists of a namenode and a
+/// datanode. We will send rpc to namenode to know which datanode to send
+/// and receive data to. Also, HDFS will rebalance data across the cluster
+/// to make sure each file has enough redundancy.
+///
+/// ```txt
+///                     ┌───────────────┐
+///                     │  Data Node 2  │
+///                     └───────────────┘
+///                             ▲
+/// ┌───────────────┐           │            ┌───────────────┐
+/// │  Data Node 1  │◄──────────┼───────────►│  Data Node 3  │
+/// └───────────────┘           │            └───────────────┘
+///                     ┌───────┴───────┐
+///                     │   Name Node   │
+///                     └───────────────┘
+///                             ▲
+///                             │
+///                      ┌──────┴─────┐
+///                      │   Vector   │
+///                      └────────────┘
+/// ```
+///
+/// For more information, please refer to:
+///
+/// - [HDFS Users Guide](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsUserGuide.html)
+/// - [opendal::services::hdfs](https://docs.rs/opendal/latest/opendal/services/struct.Hdfs.html)
+#[configurable_component(sink("hdfs"))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+
+pub struct HdfsConfig {
+    /// The root path of hdfs services.
+    ///
+    /// Root must be a valid dir.
+    ///
+    /// The final file path with be like `{root}/{prefix}{suffix}`.
+    #[serde(default)]
+    pub root: String,
+    /// A prefix to apply to all keys.
+    ///
+    /// Prefixes are useful for partitioning objects, such as by creating an blob key that
+    /// stores blobs under a particular "directory". If using a prefix for this purpose, it must end
+    /// in `/` to act as a directory path. A trailing `/` is **not** automatically added.
+    ///
+    /// The final file path with be like `{root}/{prefix}{suffix}`.
+    #[serde(default)]
+    #[configurable(metadata(docs::templateable))]
+    pub prefix: String,
+
+    /// An HDFS cluster consists of a single NameNode, a master server that manages the file system namespace and regulates access to files by clients.
+    ///
+    /// For example:
+    ///
+    /// - `default`: visiting local fs.
+    /// - `http://172.16.80.2:8090` visiting name node at `172.16.80.2`
+    ///
+    /// For more information: [HDFS Architecture](https://hadoop.apache.org/docs/r3.3.4/hadoop-project-dist/hadoop-hdfs/HdfsDesign.html#NameNode_and_DataNodes)
+    #[configurable(derived)]
+    #[serde(default)]
+    pub name_node: String,
```

Comment:
```suggestion
    #[serde(default)]
    #[configurable(metadata(docs::examples = "default")]
    #[configurable(metadata(docs::examples = "http://172.16.80.2:8090")]
    pub name_node: String,
```

**suggestion**: Adding examples for the documentation, rather than in the description.

Also, I'm not too familiar with HDFS - is `default` a DNS style name or something else? If the NameNode is remote is a full address (including protocol, ip (dns name?), and port required?

Reply:
Oh, I made a mistake here. The namenode should point to a valid hdfs path. `default` points to the `default` section specified in the Hadoop deployment configuration. In most cases, user specify the namenode via full hdfs address like `hdfs://127.0.0.1:9000`.

I will fix it.